skops/io/tests/test_persist.py

```diff
@@ -804,3 +804,30 @@ def test_loads_from_str():
     msg = "Can't load skops format from string, pass bytes"
     with pytest.raises(TypeError, match=msg):
         loads("this is a string")
+
+
+class _BoundMethodHolder:
+    """Used to test the ability to serialise and deserialize bound methods"""
+
+    def __init__(self, variant: str):
+        if variant == "sqrt":
+            self.chosen_function = np.sqrt
+        elif variant == "log":
+            self.chosen_function = np.log
+        else:
+            self.chosen_function = np.exp
+
+    def apply(self, x):
+        return self.chosen_function(x)
+
+
+def test_for_serialized_bound_method_works_as_expected(tmp_path):
+    obj = _BoundMethodHolder(variant="log")
+    bound_function = obj.apply
+    transformer = FunctionTransformer(func=bound_function)
+
+    loaded_transformer = save_load_round(transformer, tmp_path / "file.skops")
+    loaded_bound_function = loaded_transformer.func.__self__.chosen_function
+
+    assert loaded_transformer.func.__name__ == bound_function.__name__
+    assert loaded_bound_function == obj.chosen_function
```

Comment:
Should we add `assert_params_equal(loaded_transformer.__dict__, transformer.__dict__)` for good measure?

Reply:
I think this would make a lot of sense to add, especially if moving towards a more clear test as in your above comment